Urban-Fun/View Controllers/ProfileViewController.m

```diff
@@ -79,4 +109,18 @@ - (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
 }
 */
 
+- (nonnull __kindof UICollectionViewCell *)collectionView:(nonnull UICollectionView *)collectionView cellForItemAtIndexPath:(nonnull NSIndexPath *)indexPath {
+    ActivityCollectionCell *cell = [self.collectionView dequeueReusableCellWithReuseIdentifier:@"activityCell" forIndexPath:indexPath];
+    Activity *activity = self.activitiesByUser[indexPath.row];
+    [cell.activityImage setImageWithURL:[NSURL URLWithString:activity.image.url]];
+    
+    return cell;
+}
+
+- (NSInteger)collectionView:(nonnull UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
+    return self.profileToView.activitiesHosted.count;
+}
+
+
```

Comment:
Another thing in this file, we can enable the navigation when the user tap the cell into the activity, since you already have everything here you only need to implement the method to handle the tap 😀